packages/agreement-process/test/submitAgreement.test.ts

```diff
@@ -0,0 +1,1742 @@
+    });
+
+    expect(nonArchivedAgreementEvent).toMatchObject({
+      type: "AgreementAdded",
+      event_version: 2,
+      version: "0",
+      stream_id: relatedAgreements.nonArchivableRelatedAgreement.id,
+    });
+  }
+
+  it("should throw an agreementNotFound when Agreement not found", async () => {
+    await addOneAgreement(getMockAgreement());
+
+    const agreementId = generateId<AgreementId>();
+    const authData = getRandomAuthData();
+    await expect(
+      agreementService.submitAgreement(
+        agreementId,
+        { consumerNotes: "This is a test" },
+        {
+          authData,
+          correlationId: randomUUID(),
+          serviceName: "AgreementServiceTest",
+          logger: genericLogger,
+        }
+      )
+    ).rejects.toThrowError(agreementNotFound(agreementId));
+  });
+
+  it("should throw an operationNotAllowed error when requester is not consumer", async () => {
+    const consumer = getMockTenant();
+    const producer = getMockTenant();
+
+    const agreement = getMockAgreement(
+      generateId<EServiceId>(),
+      consumer.id,
+      agreementState.draft
+    );
+
+    await addOneAgreement(agreement);
+
+    const authData = getRandomAuthData(producer.id);
+
+    await expect(
+      agreementService.submitAgreement(
+        agreement.id,
+        { consumerNotes: "This is a test" },
+        {
+          authData,
+          correlationId: randomUUID(),
+          serviceName: "AgreementServiceTest",
+          logger: genericLogger,
+        }
+      )
+    ).rejects.toThrowError(operationNotAllowed(authData.organizationId));
+  });
+
+  it("should throw an agreementNotInExpectedState error when have not submittable state", async () => {
+    const consumer = getMockTenant();
+    const producer = getMockTenant();
+
+    const agreement = {
+      ...getMockAgreement(
+        generateId<EServiceId>(),
+        consumer.id,
+        randomArrayItem(agreementSubmissionConflictingStates)
+      ),
+      producerId: producer.id,
+    };
+    await addOneAgreement(agreement);
+
+    const authData = getRandomAuthData(consumer.id);
+
+    await expect(
+      agreementService.submitAgreement(
+        agreement.id,
+        { consumerNotes: "This is a test" },
+        {
+          authData,
+          correlationId: randomUUID(),
+          serviceName: "AgreementServiceTest",
+          logger: genericLogger,
+        }
+      )
+    ).rejects.toThrowError(
+      agreementNotInExpectedState(agreement.id, agreement.state)
+    );
+  });
+
+  it("should throw an agreementAlreadyExists error when agreement already exists", async () => {
+    const consumer = getMockTenant();
+    const producer = getMockTenant();
+
+    const agreement = {
+      ...getMockAgreement(generateId<EServiceId>(), consumer.id),
+      producerId: producer.id,
+    };
+
+    const pendingAgreement = {
+      ...getMockAgreement(),
+      state: agreementState.pending,
```

Comment:
`agreementSubmissionConflictingStates` can also be `missingCertifiedAttributes`, you could make this random to test both cases:
```suggestion
      state: randomArrayItem(agreementSubmissionConflictingStates),
```